feature-libs/commerce-quotes/components/details/overview/commerce-quotes-details-overview.component.spec.ts

```diff
@@ -0,0 +1,125 @@
+
+const mockCartId = '1234';
+const mockQuote: Quote = {
+  allowedActions: ['EDIT'],
+  cartId: mockCartId,
+  code: '00001233',
+  creationTime: new Date('2022-06-07T11:45:42+0000'),
+  description: 'Quote Description',
+  expirationTime: new Date('2022-07-07T23:59:59+0000'),
+  updatedTime: new Date('2022-06-09T13:31:36+0000'),
+  previousEstimatedTotal: {
+    currencyIso: 'USD',
+    formattedValue: '$0.00',
+    value: 0,
+  },
+  state: 'BUYER_ORDERED',
+};
+
+class MockCommerceQuotesFacade implements Partial<CommerceQuotesFacade> {
+  getQuoteDetails(): Observable<Quote> {
+    return of(mockQuote);
+  }
+  setSort = createSpy();
+  setCurrentPage = createSpy();
+}
+
+class MockTranslationService implements Partial<TranslationService> {
+  translate(key: string): Observable<string> {
+    return of(key);
+  }
+}
+
+describe('CommerceQuotesDetailsOverviewComponent', () => {
+  let fixture: ComponentFixture<CommerceQuotesDetailsOverviewComponent>;
+  let component: CommerceQuotesDetailsOverviewComponent;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [I18nTestingModule, CardModule, RouterTestingModule],
+      declarations: [CommerceQuotesDetailsOverviewComponent],
+      providers: [
+        {
+          provide: CommerceQuotesFacade,
+          useClass: MockCommerceQuotesFacade,
+        },
+        { provide: TranslationService, useClass: MockTranslationService },
+      ],
+    }).compileComponents();
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(CommerceQuotesDetailsOverviewComponent);
+    component = fixture.componentInstance;
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should display overview if it is available', () => {
+    //when
+    fixture.detectChanges();
+
+    //then
+    const quoteOverviewElement = fixture.debugElement.query(
+      By.css('.cx-quote-overview')
+    );
+    expect(quoteOverviewElement.nativeElement.innerHTML).toBeDefined();
+  });
+
+  it('should display titles and content in card if details are available', () => {
+    //when
+    fixture.detectChanges();
+
+    //then
+    const titleElements = fixture.debugElement.queryAll(
+      By.css('.cx-card-title')
+    );
+    const contentElements = fixture.debugElement.queryAll(
+      By.css('.cx-card-label')
+    );
+    const descriptionElement = fixture.debugElement.query(
+      By.css('.truncated-text')
+    );
+    expect(titleElements.length).toEqual(7);
+    expect(contentElements.length).toEqual(6);
+    expect(descriptionElement.nativeElement.innerHTML).toBeDefined();
+  });
+
+  it('should return object with title and text if value is defined when getCardContent', (done) => {
+    //given
+    const value = 'test';
+    const titleKey = 'key';
+    const expected = { title: 'key', text: [value] };
+
+    //then
+    component.getCardContent(value, titleKey).subscribe((result) => {
+      expect(result).toEqual(expected);
+    });
+    done();
```

Comment:
I think this `done()` should be inside about subscribe arrow function

Reply:
Good catch! Actually I can omit `done` callback because stream will emit only one value and I'm not using it outside its scope as I have assertion inside `subscribe`. I'll remove it.